src/kashinath/directStateMutation.js

```diff
@@ -0,0 +1,11 @@
+function MyComponent() {
+    const [state, setState] = React.useState({ count: 0 });
+
+    const increment = () => {
+        state.count += 1; 
+        setState(state); 
+    };
+
```

Comment:
**Avoiding Direct State Mutation in React Original Code:**

- Directly mutating the state (e.g., state.count += 1) prevents React from detecting the change because the state object reference stays the same. React relies on immutability to recognize when state has changed and trigger a re-render.
- Using the functional form of setState creates a new state object based on the previous state, which allows React to detect the change and update the UI correctly. This method ensures accurate state management and improves component performance by enabling efficient re-rendering.

`const increment = () => {
        setState(prevState => ({ count: prevState.count + 1 }));
`